code/game/gamemodes/demon/demoninfo.dm

```diff
@@ -0,0 +1,474 @@
+#define POWERUPTHRESHOLD 3 //How many souls are needed per stage.
+
+#define BASIC_DEMON 0
+#define BLOOD_LIZARD 1
+#define TRUE_DEMON 2
+#define ARCH_DEMON 3
+
+#define SOULVALUE soulsOwned.len-reviveNumber
+
+#define DEMONRESURRECTTIME 600
+
+var/list/allDemons = list()
+
+/datum/demoninfo/
+	var/datum/mind/owner = null
+	var/obligation
+	var/ban
+	var/bane
+	var/banish
+	var/truename
+	var/list/datum/mind/soulsOwned = new
+	var/reviveNumber = 0
+	var/form = BASIC_DEMON
+
+/proc/randomDemonInfo(var/name = randomDemonName())
+	var/datum/demoninfo/demon = new
+	demon.truename = name
+	demon.bane = randomdemonbane()
+	demon.obligation = randomdemonobligation()
+	demon.ban = randomdemonban()
+	demon.banish = randomdemonbanish()
+	return demon
+
+/proc/demonInfo(var/name, var/saveDetails = 1)
+	if(allDemons[name])
+		return allDemons[name]
+	else
+		var/datum/demoninfo/demon = randomDemonInfo(name)
+		allDemons[name] = demon
+		return demon
+
+
+
+/proc/randomDemonName()
+	var/preTitle = ""
+	var/title = ""
+	var/mainName = ""
+	var/suffix = ""
+	if(prob(65))
+		if(prob(35))
+			preTitle = pick("Dark ", "Hellish ", "Fiery ", "Sinful ", "Blood ")
+		title = pick("Lord ", "Fallen Prelate ", "Count ", "Viscount ", "Vizier ", "Elder ", "Adept ")
+	var/probability = 100
+	mainName = pick("Hal", "Ve", "Odr", "Neit", "Ci", "Quon", "Mya", "Folth", "Wren", "Gyer", "Geyr", "Hil", "Niet", "Twou", "Hu", "Don")
+	while(prob(probability))
+		mainName += pick("hal", "ve", "odr", "neit", "ca", "quon", "mya", "folth", "wren", "gyer", "geyr", "hil", "niet", "twoe", "phi", "coa")
+		probability -= 20
+	if(prob(40))
+		suffix = pick(" the Red", " the Soulless", " the Master", ", the Lord of all things", ", Jr.")
+	return preTitle + title + mainName + suffix
+
+/proc/randomdemonobligation()
+	return pick(OBLIGATION_FOOD, OBLIGATION_DRINK, OBLIGATION_GREET, OBLIGATION_PRESENCEKNOWN, OBLIGATION_SAYNAME, OBLIGATION_ANNOUNCEKILL, OBLIGATION_ANSWERTONAME)
+
+/proc/randomdemonban()
+	return pick(BAN_HURTWOMAN, BAN_CHAPEL, BAN_HURTPRIEST, BAN_AVOIDWATER, BAN_STRIKEUNCONCIOUS, BAN_HURTLIZARD, BAN_HURTANIMAL)
+
+/proc/randomdemonbane()
+	return pick(BANE_SALT, BANE_LIGHT, BANE_IRON, BANE_WHITECLOTHES, BANE_SILVER, BANE_HARVEST, BANE_TOOLBOX)
+
+/proc/randomdemonbanish()
+	return pick(BANISH_WATER, BANISH_COFFIN, BANISH_FORMALDYHIDE, BANISH_RUNES, BANISH_CANDLES, BANISH_DESTRUCTION, BANISH_FUNERAL_GARB)
+
+/datum/demoninfo/proc/obligationlaw()
```

Comment:
Just change those to not be numbers? those defines are only used (as far as I've seen) as organisational defines (they defeat typos, and give meaning to magic numbers), which means you can just change the values of those defines to be anything, even the exact same text like ```#define OBLIGATION_FOOD "obligation food"```